dGame/dComponents/RocketLaunchpadControlComponent.cpp

```diff
@@ -143,22 +79,20 @@ void RocketLaunchpadControlComponent::Launch(Entity* originator, LWOOBJID option
 	
 	GameMessages::SendFireEventClientSide(m_Parent->GetObjectID(), originator->GetSystemAddress(), u"RocketEquipped", rocket->GetId(), cloneId, -1, originator->GetObjectID());
 
-	rocket->Equip(true);
-	
 	GameMessages::SendChangeObjectWorldState(rocket->GetId(), WORLDSTATE_ATTACHED, UNASSIGNED_SYSTEM_ADDRESS);
 	
 	EntityManager::Instance()->SerializeEntity(originator);
 }
 
 void RocketLaunchpadControlComponent::OnUse(Entity* originator) {
-	auto* propertyEntrance = m_Parent->GetComponent<PropertyEntranceComponent>();
+	auto* rocket = originator->GetComponent<CharacterComponent>()->RocketEquip(originator);
+	if (!rocket) return;
 
-	if (propertyEntrance != nullptr)
-	{
-		propertyEntrance->OnUse(originator);
+	auto* propertyEntrance = m_Parent->GetComponent<PropertyEntranceComponent>();
```

Comment:
Would put a comment here how this returns early because the launch is handled within their respective components.